app/utils/sagaInjectors.js

```diff
@@ -25,39 +25,46 @@ const checkDescriptor = descriptor => {
 
 export function injectSagaFactory(store, isValid) {
   return function injectSaga(key, descriptor = {}, args) {
-    if (!isValid) checkStore(store);
+    return new Promise((resolve, reject) => {
+      try {
+        if (!isValid) checkStore(store);
 
-    const newDescriptor = {
-      ...descriptor,
-      mode: descriptor.mode || DAEMON,
-    };
-    const { saga, mode } = newDescriptor;
+        const newDescriptor = {
+          ...descriptor,
+          mode: descriptor.mode || RESTART_ON_REMOUNT,
```

Comment:
The default mode has been switched to DAEMON in #2430. Please make sure you don't change it back to RESTART_ON_REMOUNT as it appears here.

Reply:
I know the default mode has been switched to DAEMON, and in #2121, to avoid the `componentWillMount` issue, it changed to DAEMON.

But the Pull Request is to fix `componentWillMount` issue

Reply:
It was also changed to DAEMON because most people found it more intuitive. Please leave it that way.

If it must be changed again, it should be done in a separate PR after we've had a discussion about it and agreed that there's a more intuitive choice than DAEMON.